### PR TITLE
storybook(fxa-settings): Recreate SigninTotpCode in React

### PR DESCRIPTION
**Pull request by vpomerleau**
## Because

- We're converting the remaining content-server Backbone views to React. The first step is to recreate them in React, and house them in FxA Settings' storybook until they're ready to be attached to functionality.

## This pull request

- Recreate the SigninTotpCode view in React, with the initial metrics, l10n, and tests.
- Update SigninTokenCode (add missing l10n)

## Issue that this pull request solves

Closes: #FXA-6422

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

Original content-server view:
image

New storybook:
image

## Other information (Optional)

Any other information that is important to this pull request.

### CODE REVIEW

File: packages/fxa-settings/src/pages/SigninTokenCode/en.ftl
Comment (bcolsson) [inline packages/fxa-settings/src/pages/SigninTokenCode/en.ftl:10]: Even when changing or adding an attribute to an existing string, you'll need to make a new ID for the string.

File: packages/fxa-settings/src/pages/SigninTotpCode/en.ftl
Comment (bcolsson) [inline packages/fxa-settings/src/pages/SigninTotpCode/en.ftl:1]: The "totp" here is a bit too technical and might not be clear to localizers (I had to look it up). This group level comment could use a bit more explanation as to what context these strings are appearing. Especially since "security code" is rather generic and some locales may have a more specific term.

File: packages/fxa-settings/src/pages/SigninTotpCode/en.ftl
Comment (bcolsson) [inline packages/fxa-settings/src/pages/SigninTotpCode/en.ftl:17]: Previous strings are using `security code` while this is using `authentication code`. 

Recently a lot of the messaging around two factor backup codes were unified under the term `backup authentication code` so it seems like it might be better to use consistent terminology with the more specific `authentication code`.

**Review by bcolsson — CHANGES_REQUESTED** [review]

File: packages/fxa-settings/src/pages/SigninTokenCode/en.ftl
Comment (vpomerleau) [reply packages/fxa-settings/src/pages/SigninTokenCode/en.ftl:10]: Meant to change that. Sorry! I'll update the ID to `signin-token-code-input-label`

File: packages/fxa-settings/src/pages/SigninTotpCode/en.ftl
Comment (vpomerleau) [reply packages/fxa-settings/src/pages/SigninTotpCode/en.ftl:1]: Sure thing, I can add more context to the group comment to explain when a user would see this page.

File: packages/fxa-settings/src/pages/SigninTotpCode/en.ftl
Comment (vpomerleau) [reply packages/fxa-settings/src/pages/SigninTotpCode/en.ftl:17]: I agree that there are currently inconsistencies in the terminology, but the strings used here match those is currently in production and already localized (`client.po`).

@LZoog  - @bcolsson made some very good observations that the language used for different types of codes in `SigninTokenCode`, `SigninTotpCode` and `SigninRecoveryCode` is inconsistent and possibly confusing for users. Auditing our terminology would probably fit it with some of our OKRs for 2023, but might not be within scope for these storybook tickets. I'd suggest keeping the existing strings for the moment, and filing a spike(?) to review messaging.

Comment (bcolsson) [reply packages/fxa-settings/src/pages/SigninTotpCode/en.ftl:20]: It would be more efficient to address these strings as they arise since as we move strings to fluent new translation requests are going to be triggered.

That being said, since "security code" is a pretty broad term and it has been a while since these strings were last translated, even if en-US strings don't change there's a possibility that translations may diverge. If we do go this route, it'd be best to have some additional comments for any string with "security code" to describe what type of code (two factor authentication code) this is.

Comment (LZoog) [reply packages/fxa-settings/src/pages/SigninTotpCode/en.ftl:20]: We're calling this "security code" similarly on our Settings page. If we change this here we should change it there as well. https://github.com/mozilla/fxa/blob/main/packages/fxa-settings/src/components/Settings/PageTwoStepAuthentication/en.ftl#L34

@vpomerleau What's that ticket number? We could escalate this to Product given the conversation here and we may want to leave this string unlocalized for now with a ticket attached while the decision is made. It's probably going to be a little while before this page is hooked up and available to all users. FWIW, we recently had a related conversation around recovery key vs  code in FXA-4423.

File: packages/fxa-settings/src/pages/SigninTotpCode/index.tsx
Comment (LZoog) [inline packages/fxa-settings/src/pages/SigninTotpCode/index.tsx:90]: All `role="img"` SVGs should be given `aria-label`s which will work like alt text. If content-server doesn't provide it, we can make our best guess on what describes it and if we're not sure we can run it by Vesta. Maybe this can be, "Device with hidden code" or something like that.

File: packages/fxa-settings/src/pages/SigninTotpCode/index.tsx
Comment (LZoog) [inline packages/fxa-settings/src/pages/SigninTotpCode/index.tsx:74]: I wonder if we'll want to use the alert bar on these pages vs the error div that usually sits within the cards themselves, hmm.

**Review by LZoog — APPROVED** [review]
LGTM, pending Bryan's approval on how we handle that string. I think it's OK for now to leave it unlocalized if that's what we want to do.

Couple nits but I don't think it's worth blocking r+, but also don't forget to add the events here to the [spreadsheet](https://docs.google.com/spreadsheets/d/1na5BqUNefrgsK-7wTlKoy-GuacLZ0rfSZ1bLiJkwFOk/). 😁

File: packages/fxa-settings/src/pages/SigninTotpCode/en.ftl
Comment (bcolsson) [reply packages/fxa-settings/src/pages/SigninTotpCode/en.ftl:20]: There's no way to leave it unlocalized if we're going to commit the PR with these strings in them (there's no way to cherry-pick strings not to put into Pontoon). If we want to get this PR merged, then the best route would to include comments for strings that contain "security code" to explain what type of code it is.

Comment (vpomerleau) [reply packages/fxa-settings/src/pages/SigninTotpCode/en.ftl:20]: @bcolsson Part of the question here is  - are we thinking of replacing "security code" or "two-step authentication code"? If "two step authentication" in the error message is the one we'd like to standardize, then I would remove that one string from the FTL file and leave a note in the code to revisit this string and add l10n once we have product's recommendation.

Comment (LZoog) [reply packages/fxa-settings/src/pages/SigninTotpCode/en.ftl:20]: > There's no way to leave it unlocalized if we're going to commit the PR with these strings in them (there's no way to cherry-pick strings not to put into Pontoon). If we want to get this PR merged, then the best route would to include comments for strings that contain "security code" to explain what type of code it is.

We should be able to leave it unlocalized if we don't add the FTL ID and string to the FTL file and the implementation is taken out in the TSX. 🙂

Comment (bcolsson) [reply packages/fxa-settings/src/pages/SigninTotpCode/en.ftl:20]: Ah, was assuming that wasn't an option, but yes if you're okay with that route that's perfectly fine for l10n 😄

Comment (LZoog) [reply packages/fxa-settings/src/pages/SigninTotpCode/en.ftl:20]: > Ah, was assuming that wasn't an option, but yes if you're okay with that route that's perfectly fine for l10n 😄

Yeah, since these pages won't be user-facing for a bit I think we can have a follow up ticket to localize it once we're sure what the string should be to not block merging this 👍

File: packages/fxa-settings/src/pages/SigninTotpCode/index.tsx
Comment (vpomerleau) [reply packages/fxa-settings/src/pages/SigninTotpCode/index.tsx:90]: @LZoog I feel this image is decorative (doesn't add information to the page) and doesn't need alt text, as described here: https://www.w3.org/WAI/tutorials/images/decorative/
Does the team prefer to add alt-text to all images, decorative or not? Also, I understand that wheter or not this image is decorative might be subjective.

File: packages/fxa-settings/src/pages/SigninTotpCode/index.tsx
Comment (vpomerleau) [reply packages/fxa-settings/src/pages/SigninTotpCode/index.tsx:74]: Could be up for future decision, when we hook up functionality and test?

File: packages/fxa-settings/src/pages/SigninTotpCode/index.stories.tsx
Comment (vpomerleau) [reply packages/fxa-settings/src/pages/SigninTotpCode/index.stories.tsx:13]: I like the idea. 

@millsoper we should do this for all `Signin` PRs that are currently up for review. Perhaps you could move the already merged `Signin` folders in the last remaining ticket for this epic? (FXA-6352)?

File: packages/fxa-settings/src/pages/SigninTotpCode/index.tsx
Comment (vpomerleau) [reply packages/fxa-settings/src/pages/SigninTotpCode/index.tsx:106]: I added a comment above the input here for future consideration.

File: packages/fxa-settings/src/pages/SigninTotpCode/en.ftl
Comment (vpomerleau) [reply packages/fxa-settings/src/pages/SigninTotpCode/en.ftl]: I've removed that error string from the FTL file, and commented out the FtlMsg code from the index file. 

Also pinged Vesta on the [Jira ticket](https://mozilla-hub.atlassian.net/browse/FXA-6422) so we can update the string in a later ticket once we have her input.

File: packages/fxa-settings/src/pages/SigninTokenCode/en.ftl
Comment (bcolsson) [reply packages/fxa-settings/src/pages/SigninTokenCode/en.ftl:12]: Double checking just in case but the .label attribute has been deleted.

**Review by bcolsson — APPROVED** [review]

File: packages/fxa-settings/src/pages/SigninTokenCode/en.ftl
Comment (vpomerleau) [reply packages/fxa-settings/src/pages/SigninTokenCode/en.ftl:12]: Thanks for double checking, spotted an error in this update - latest push has fixed ids.

Comment (bcolsson) [reply packages/fxa-settings/src/pages/SigninTokenCode/en.ftl:12]: See the new IDs now, looks good 👍

**Pull request merged**
